### ISSUE TITLE
Critical fault by entering admin or checkout

### ISSUE THREAD

**Issue by johnrootk** [wallee-payment/woocommerce#26]
Good morning,

Today the plugin was updated from version 1.7.18 to 2.0.6 on our website. Since then, we are no longer able to access the admin dashboard. Following error we receive:
WordPress-Version 6.0.3
Aktuelles Plugin: wallee (Version 2.0.6) PHP-Version 7.4.32

Fehler-Details
==============
Ein Fehler vom Typ E_ERROR wurde in der Zeile 424 der Datei /home/httpd/vhosts/partyup.ch/httpdocs/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php verursacht. Fehlermeldung: WC_Wallee_Helper::get_clean_locale_for_string(): The script tried to execute a method or access a property of an incomplete object. Please ensure that the class definition "Wallee\Sdk\Model\RestLanguage" of the object you are trying to operate on was loaded _before_ unserialize() gets called or provide an autoloader to load the class definition

The frontpage is currently working. But the checkout page is failing also. I hope for a fast fix.

In the meanwhile, I downgraded to 1.7.23

Best regards,
John

**Comment by domi-87** [comment]
Same here:

WordPress-Version 6.0.3
Aktuelles Plugin: wallee (Version 2.0.6)
PHP-Version 8.1.11



Fehler-Details
==============
Ein Fehler vom Typ E_ERROR wurde in der Zeile 424 der Datei /var/www/vhosts/[domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php](http://domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php) verursacht. Fehlermeldung: Uncaught Error: The script tried to call a method on an incomplete object. Please ensure that the class definition "Wallee\Sdk\Model\RestLanguage" of the object you are trying to operate on was loaded _before_ unserialize() gets called or provide an autoloader to load the class definition in /var/www/vhosts/[domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php:424](http://domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php:424)
Stack trace:
#0 /var/www/vhosts/[domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php(385)](http://domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php(385)): WC_Wallee_Helper->get_clean_locale_for_string()
#1 /var/www/vhosts/[domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php(122)](http://domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-helper.php(122)): WC_Wallee_Helper->get_cleaned_locale()
#2 /var/www/vhosts/[domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-gateway.php(92)](http://domain/wp-content/plugins/woo-wallee/includes/class-wc-wallee-gateway.php(92)): WC_Wallee_Helper->translate()
#3 /var/www/vhosts/[domain/wp-content/plugins/woo-wallee/woocommerce-wallee.php(624)](http://domain/wp-content/plugins/woo-wallee/woocommerce-wallee.php(624)): WC_Wallee_Gateway->__construct()
#4 /var/www/vhosts/[domain/wp-includes/class-wp-hook.php(307)](http://domain/wp-includes/class-wp-hook.php(307)): WooCommerce_Wallee->add_gateways()
...
#17 {main}
thrown

**Comment by theodorspringwallee** [comment]
Dear [johnrootk](https://github.com/johnrootk), [domi-87](https://github.com/domi-87)

We apologize for these circumstances. 

If you are accessing your shop via (S)FTP you can rename our plugin, this should help to access the admin backend again.

We will check this BUG from our side as soon as possible. 

Like already mentioned by you, it's the best behavior to downgrade the wallee plugin for now.

We will let you know, when this BUG is fixed and released

Best regards

**Comment by theodorspringwallee** [comment]
Dear [johnrootk](https://github.com/johnrootk), [domi-87](https://github.com/domi-87)

The mentioned issue should be fixed now in our newest release: https://github.com/wallee-payment/woocommerce/releases/tag/2.0.8

Best regards

**Comment by theodorspringwallee** [comment]
Dear [johnrootk](https://github.com/johnrootk), [domi-87](https://github.com/domi-87)

Please ignore our previous comment, there seems to be still an issue there.

We will let you know, when the issue is finally fixed.

Best regards

**Comment by theodorspringwallee** [comment]
Dear [johnrootk](https://github.com/johnrootk), [domi-87](https://github.com/domi-87)

We want to inform you about our previous investigations:

- We tested your mentioned issue on multiple of our test instances.
- On one (docker based) test instance we could replicate this issue. 
- Therefore, we could solve the issue, by re-start this server.
- On all other instances with the same versions we couldn't replicate this issue.

According to our developers this issue is a (sporadic) session or cache issue, that is related with some renewed serialization logics within our plugin code, for that in that occurred case was used with wrong data, because of some old temporary saved data within the shop system, that raised this serialization issue.

**This issue could be solved on our side, by restarting the server.**

Therefore, we kindly ask if it would be possible for you to restart your server and check if the issue with the newest plugin version still occur.

In the meanwhile we try on our side to still find some replication possibilities on our test instances, to prevent this issue, that is coming sporadically with our newest plugin version.

Please let us know if you have further question on this.

Best regards

**Comment by johnrootk** [comment]
Hi @theodorspringwallee 

Thank you for your investigation. I updated now to version 2.0.8 and cleared all caching (Page-Cache and Server). It seems to work now. 
I am able to access all pages, including checkout without an error. 

Thank you and have a nice day!

**Comment by theodorspringwallee** [comment]
Hi [johnrootk](https://github.com/johnrootk)

Thank you very much for your feedback.

We would appreciate if you can tell us, how exactly cleared the cache.

Did you via CLI with `wp cache flush` and separately some server cache (also PHP cache?).

We really would appreciate your feedback, that we can provide a good installation workaround for other customers with this issue.

Many thanks & best regards

**Comment by johnrootk** [comment]
Hi @theodorspringwallee 

After the update, the error happened again. So I entered my admin page in safe-mode.
We use WPRocket for page caching. So I cleared and reload their cache in the admin backend. 
This already seems to fix the problem. 
I still entered my webserver and cleared the opcache, to be sure. 

Btw. with the new version, I always see:
```
0 manuelle Tasks benötigen Ihre Aufmerksamkeit.

[Ansicht]
```
which I think it shouldn't be visible on 0 tasks? Nothing critical.
